### PR TITLE
MOO-60: Full patient record

### PR DESCRIPTION
**Pull request by Imran536**
PR for MOO-60

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/msfcore/api/impl/PatientSummaryServiceImpl.java [hunk 1]
```diff
@@ -311,70 +376,168 @@ private void setClinicalHistory(Patient patient, PatientSummary patientSummary)
...
+                if (ob.getConcept().getConceptId() == 6000009) {
+                    Observation o = convertObs(ob);
+                    appointment.setType(o.getValue());
+                } else if (ob.getConcept().getConceptId() == 463411) {
+                    Observation o = convertObs(ob);
+                    appointment.setDate(o.getValue());
+                }
+            }
+            if (appointment.getType() != null || appointment.getDate() != null)
+                patientSummary.getAppointmentList().add(appointment);
```
Comment (edrisse) [inline api/src/main/java/org/openmrs/module/msfcore/api/impl/PatientSummaryServiceImpl.java]: Please nest all the single instruction if or else statements inside {}

File: api/src/main/java/org/openmrs/module/msfcore/api/impl/PatientSummaryServiceImpl.java [hunk 2]
```diff
@@ -388,4 +551,36 @@ public PatientSummary requestPatientSummary(Patient patient) {
...
+        int encTypeId = 0;
+        for (Encounter encounter : Context.getEncounterService().getEncountersByPatient(patient)) {
+            if (encounter.getEncounterType().getEncounterTypeId() != encTypeId) {
+                org.openmrs.module.msfcore.patientSummary.Encounter enc = org.openmrs.module.msfcore.patientSummary.Encounter.builder()
+                                .build();
+                enc.setDate(Context.getDateFormat().format(encounter.getEncounterDatetime()));
+                enc.setType(encounter.getEncounterType().getName());
+                try {
+                    enc.setProvider(((EncounterProvider) encounter.getEncounterProviders().iterator().next()).getProvider().getName());
+                } catch (Exception e) {
```
Comment (edrisse) [inline api/src/main/java/org/openmrs/module/msfcore/api/impl/PatientSummaryServiceImpl.java]: Please handle the specific exception we are looking of here

File: api/src/main/java/org/openmrs/module/msfcore/MSFCoreActivator.java [hunk 3]
```diff
@@ -44,9 +44,11 @@
...
     public void started() {
         log.info("Started MSF Core Module");
-
-        Context.getService(DHISService.class).transferDHISMappingsToDataDirectory();
-        Context.getService(DHISService.class).installDHIS2Metadata();
+        try {
+            Context.getService(DHISService.class).transferDHISMappingsToDataDirectory();
+            Context.getService(DHISService.class).installDHIS2Metadata();
+        } catch (Exception e) {
+        }
```
[the remark this answers is not in the document]
Comment (lauravignoli) [reply api/src/main/java/org/openmrs/module/msfcore/MSFCoreActivator.java]: @Imran536 I recommend merging master to this branch, as I added this check there already.

**Comment by edrisse** [comment]
@Imran536 unit tests are failing. Please fix.

**Comment by edrisse** [comment]
@Imran536 also I found the following problems on the screen:
- Diagnosis Details table still does not have headers;
- In Encounters table, sort rows by date desc;
- In Medications table, add dispensed and status columns;
- Order sections as per mockup;
- Add footer;

**Pull request closed without merging**